### PR TITLE
generic improvement for project

### PR DESCRIPTION
**Pull request by mba-odoo**
Purpose of the commit is to do the generic improvement for the refactoring of project.

So in this commit, done the following changes:
 - move the date_deadline and the tag_ids field to the right column
 - remove the warning 'by saving this change, the customer email and phone
number will also be updated.'
 - track the priority field on task
 - move the deadline field above the fa-star and fa-clock-o icons
 - the name of archived tasks should be crossed out
 - project form add a 'deadline' field below the user_id one
 - project form remove the 'description' notebook
 - remove the 'project' field from the quick create of task.

web:
 - Display the label with state_selection widget icon in list view. user
can hide the the label with option 'hide_label'

TaskID: 2451287

### CODE REVIEW

File: addons/web/static/src/js/fields/basic_fields.js [hunk 1]
```diff
@@ -2599,6 +2599,15 @@ var StateSelectionWidget = AbstractField.extend({
...
 
+        // Add label with status icon
+        if (this.nodeOptions && this.nodeOptions.show_label) {
+            const $label = $('<span>', {
+                text: currentState.state_name,
+                class: 'align-middle',
+            });
+            this.$el.append($label);
+        }
+
```
Comment (IT-Ideas) [inline addons/web/static/src/js/fields/basic_fields.js:2610]: Hello @mba-odoo 🙂 
As per the JS team requirement, we have to create a JS test in order to ensure that this gets not broken in the future. We will also need to update the documentation. I'll ping them in the PR.

**Review by IT-Ideas — CHANGES_REQUESTED** [review]
Hello @mba-odoo 🙂 

I still have to debate with AJU about the sheet element visual behavior.
There is sill a test and the doc to write for the web widget change. Once done I'll ping them on the PR.

Good work 🙂

File: addons/web/static/src/js/fields/basic_fields.js (hunk 1 above)
Comment (mba-odoo) [reply addons/web/static/src/js/fields/basic_fields.js:2610]: @IT-Ideas , Done the changes. Thanks

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 2]
```diff
@@ -6209,6 +6209,33 @@ QUnit.module('basic_fields', {
...
+        assert.containsN(list, '.o_state_selection_cell .o_selection > a span.o_status', 5,
+            "should have five status selection widgets");
+        assert.containsN(list, '.o_state_selection_cell .o_selection > span.align-middle', 5,
+            "should have five label on election widgets");
+        assert.containsOnce(list, '.o_state_selection_cell .o_selection > a span.o_status.o_status_green',
+            "should have one green status");
+        assert.containsOnce(list, '.o_state_selection_cell .o_selection > span.align-middle:contains(Done)',
+            "should have one Done status label");
+        assert.containsN(list, '.o_state_selection_cell .o_selection > span.align-middle:contains(Normal)', 3,
+            "should have one Normal status label");
```
Comment (IT-Ideas) [inline addons/web/static/tests/fields/basic_fields_tests.js:6234]: Shouldn't we rather only test that the label is displayed when the option is set to True and that the label is not displayed when the option is not used or when it is set to False? By doing so the test is more "unitary".

**Review by IT-Ideas — COMMENTED** [review]
Good job 🙂 
Just a small remark for the test.

Thank you 👍🏼

File: addons/web/static/tests/fields/basic_fields_tests.js (hunk 2 above)
Comment (mba-odoo) [reply addons/web/static/tests/fields/basic_fields_tests.js]: @IT-Ideas , Done all the changes. Thanks

**Comment by IT-Ideas** [comment]
Hello @mba-odoo, I have asked in Discord the JS team to review the JS framework part of the PR.

**Review by aab-odoo — CHANGES_REQUESTED** [review]
I made several remarks on the code (only reviewed the web part of course). After discussing with ngr, we agreed to slightly change the spec concerning the state_selection widget: in list views, it should display the label by default. To do that, you can create a ListStateSelectionWidget extending the StateSelectionWidget, and add it to the registry (in field_registry.js) with key `list.state_selection`. In that widget, you can support an option `hide_label` (be careful when evaluating it, cf my remark in the code).

**Comment by mba-odoo** [comment]
@aab-odoo , Done the changes. Thanks

File: doc/reference/javascript_reference.rst [hunk 3]
```diff
@@ -2149,6 +2149,19 @@ kanban_state_selection (StateSelectionWidget)
 
   - Supported field types: *selection*
 
+list.state_selection (ListStateSelectionWidget)
+  This is exactly the same widget as state_selection but display the label next to the icon
```
Comment (aab-odoo) [inline doc/reference/javascript_reference.rst:2153]: In list views, the state_selection widget displays by default the label next to the icon.

**Review by aab-odoo — COMMENTED** [review]
Could you put the web diff in a separated commit? Thanks

File: doc/reference/javascript_reference.rst (hunk 3 above)
Comment (mba-odoo) [reply doc/reference/javascript_reference.rst]: @aab-odoo , Done. thanks

**Review by aab-odoo — APPROVED** [review]
ok for the web part

**Review by IT-Ideas — APPROVED** [review]

**Comment by tivisse** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**